src/libs/SearchQueryUtils.ts

```diff
@@ -679,14 +679,13 @@ function buildFilterFormValuesFromQuery(
         if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.WITHDRAWAL_ID || filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.REPORT_ID) {
             filtersForm[key as typeof filterKey] = filterValues.join(',');
         }
-        if (
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.MERCHANT ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.DESCRIPTION ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.TITLE ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.ACTION
-        ) {
+        if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.MERCHANT || filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.DESCRIPTION || filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.TITLE) {
             filtersForm[key as typeof filterKey] = filterValues.at(0);
         }
+        if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.ACTION) {
+            const actionValue = filterValues.at(0) as ValueOf<typeof CONST.SEARCH.ACTION_FILTERS> | undefined;
```

Comment:
Hmm this cast is incorrect, right? We don't know that the filter value is one of the values in `CONST.SEARCH.ACTION_FILTERS`. Can we narrow the type down so that it's accurate and we don't need to do a cast?